pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -113,6 +115,107 @@ type ReconcileNodeNetworkConfigurationPolicy struct {
 	scheme *runtime.Scheme
 }
 
+// It will return the policies with same labels at the one from argument
+func (r *ReconcileNodeNetworkConfigurationPolicy) filterByNodeLabels(node corev1.Node) (nmstatev1alpha1.NodeNetworkConfigurationPolicyList, error) {
+	policyList := nmstatev1alpha1.NodeNetworkConfigurationPolicyList{}
+	filteredPolicyList := nmstatev1alpha1.NodeNetworkConfigurationPolicyList{}
+
+	// Retrieve all the policies
+	err := r.client.List(context.TODO(), &client.ListOptions{}, &policyList)
+	if err != nil {
+		return policyList, err
+	}
+
+	// Get the ones that fit this node
+	for _, policy := range policyList.Items {
+		if nodeSelectorMatchesThisNode(r.client, &policy) {
+			filteredPolicyList.Items = append(filteredPolicyList.Items, policy)
+		}
+	}
+
+	return filteredPolicyList, nil
+}
+
+func unmarshalInterfaces(state nmstatev1alpha1.State) (map[string]interface{}, []interface{}, error) {
+	// Unmarshall interfaces state into unstructured golang
```

Comment:
This should document the whole function. Also, please document what is being returned in the first and the second value. It looks weird that we unwrap insides and return both unstructuredState and its interfaces. Can't we handle it from the caller?